server/src/main/java/io/crate/analyze/RelationNames.java

```diff
@@ -68,6 +71,13 @@ public static SequencedSet<RelationName> getShallow(Symbol symbol) {
         return get(symbol, false);
     }
 
+    public static SequencedSet<RelationName> getDeep(AnalyzedRelation relation) {
```

Comment:
This is used to find all underlying relation names in `AnalyzedRelation`. `JoinRelations` have two sides of `AnalyzedRelation` which  can be further nested. This leads to multiple Relationnames within on `AnalyzedRelation`